sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -185,6 +185,9 @@ abstract class Optimizer(catalogManager: CatalogManager)
       RemoveLiteralFromGroupExpressions,
       RemoveRepetitionFromGroupExpressions) :: Nil ++
     operatorOptimizationBatch) :+
+    // This batch rewrites plans and should be run after the operator
+    // optimization batch and before any batches that depend on stats.
+    Batch("Rewrite Rules", Once, rewriteRules: _*) :+
```

Comment:
Basically, every optimizer rule is rewriting the plans, isn't it?

Reply:
Hmm, same question. This naming looks too general.

Reply:
Well, you are right after removing `v2Source` from the name is it a bit too abstract. Any suggestions, @dongjoon-hyun? I am not sure making it specific to v2 sources is a good idea too.

Reply:
I decided to call it `dataSourceRewriteRules` as this seems to be generic enough but yet describes what it is.

Reply:
Looks better. Thanks for the change.